### PR TITLE
add decimal support

### PR DESCRIPTION
**Pull request by akonyaev90**

### CODE REVIEW

File: src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseConverter.java [hunk 1]
```diff
@@ -33,46 +18,100 @@
...
-        map.put(CHAR, new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
-        map.put(VARCHAR, new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
-        map.put(LONGVARCHAR, new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
-        map.put(NVARCHAR, new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+        map.put(new TypeWithScaleAndPrecision(CHAR,null,null), new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+        map.put(new TypeWithScaleAndPrecision(VARCHAR,null,null), new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+        map.put(new TypeWithScaleAndPrecision(LONGVARCHAR,null,null), new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+        map.put(new TypeWithScaleAndPrecision(NVARCHAR,null,null), new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+
+        for (int precision = 1 ; precision <= 38 ; precision++ ){
```
Comment (alex-krash) [inline src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseConverter.java]: Please, use constants for magic numbers.

File: src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseDataType.java [hunk 2]
```diff
@@ -6,30 +6,49 @@
...
 
     // Misc
-    String;
+    String("String");
+
+    String typo;
 
-    public String getName(boolean isNullable) {
-        String retval = toString();
+    ClickHouseDataType(String string) {
```
Comment (alex-krash) [inline src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseDataType.java]: Maybe, let's have 2 constructors? No-argument calls single-argument with this.name().

File: src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseConverter.java (hunk 1 above)
Comment (akonyaev90) [reply src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseConverter.java]: I added DECIMAL_XX_MAXPRECISION constants

File: src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseDataType.java (hunk 2 above)
Comment (akonyaev90) [reply src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseDataType.java]: no-arguments constructor was added

**Comment by alex-krash** [comment]
Thanks, @akonyaev90 !

**Pull request merged**